File: scratch/convert.mjs
import fs from 'fs';
import path from 'path';

function convertHtmlToJsx(htmlStr) {
    // 1. Extract the body content (ignoring the scripts and head)
    let bodyMatch = htmlStr.match(/<body[^>]*>([\s\S]*?)<\/body>/i);
    let bodyContent = bodyMatch ? bodyMatch[1] : htmlStr;

    // 2. Replace class= with className=
    let jsx = bodyContent.replace(/class=/g, 'className=');

    // 3. Make sure all <img, <input, <br, <hr are self closing
    jsx = jsx.replace(/<img([^>]*[^\/])>/g, '<img$1 />');
    jsx = jsx.replace(/<input([^>]*[^\/])>/g, '<input$1 />');
    jsx = jsx.replace(/<br\s*>/g, '<br />');
    jsx = jsx.replace(/<hr\s*>/g, '<hr />');

    // 4. Transform inline styles. E.g., style="font-variation-settings: 'FILL' 1;" -> style={{ fontVariationSettings: "'FILL' 1" }}
    jsx = jsx.replace(/style="([^"]*)"/g, (match, p1) => {
        const styleObjStr = p1.split(';').filter(s => s.trim()).map(rule => {
            const [key, val] = rule.split(':').map(s => s.trim());
            if(!key) return '';
            const camelKey = key.replace(/-([a-z])/g, g => g[1].toUpperCase());
            return `${camelKey}: "${val.replace(/"/g, "'")}"`;
        }).join(', ');
        return `style={{ ${styleObjStr} }}`;
    });

    // 5. svg props fix if any (not strictly needed for material icons as they are spans)
    // 6. fix <!-- --> to {/* */}
    jsx = jsx.replace(/<!--([\s\S]*?)-->/g, '{/* $1 */}');

    return jsx;
}

const homeHtml = fs.readFileSync('home.html', 'utf8');
const facultyHtml = fs.readFileSync('faculty.html', 'utf8');

const homeJsx = convertHtmlToJsx(homeHtml);
const facultyJsx = convertHtmlToJsx(facultyHtml);

const homeComponent = `
export default function HomePage() {
  return (
    <div className="bg-background text-on-surface font-body selection:bg-primary-container selection:text-on-primary-container">
      ${homeJsx}
    </div>
  );
}
`;

const facultyComponent = `
export default function FacultyPage() {
  return (
    <div className="bg-background text-on-surface font-body selection:bg-primary-container selection:text-on-primary-container">
      ${facultyJsx}
    </div>
  );
}
`;

fs.writeFileSync('src/app/page.tsx', homeComponent.trim() + '\n');
fs.writeFileSync('src/app/faculty/page.tsx', facultyComponent.trim() + '\n');

console.log("Pages generated successfully!");

File: scratch/convert.py
import re

def convert_html_to_jsx(html_str):
    # 1. Extract body content
    body_match = re.search(r'<body[^>]*>([\s\S]*?)</body>', html_str, re.IGNORECASE)
    body_content = body_match.group(1) if body_match else html_str

    # 2. Replace class= with className=
    jsx = body_content.replace('class=', 'className=')

    # 3. Self-closing tags
    jsx = re.sub(r'<img([^>]*[^/])>', r'<img\1 />', jsx)
    jsx = re.sub(r'<input([^>]*[^/])>', r'<input\1 />', jsx)
    jsx = re.sub(r'<br\s*>', r'<br />', jsx)
    jsx = re.sub(r'<hr\s*>', r'<hr />', jsx)

    # 4. Transform inline styles
    def style_replacer(match):
        css_str = match.group(1)
        rules = [r.strip() for r in css_str.split(';') if r.strip()]
        style_obj_parts = []
        for rule in rules:
            if ':' not in rule: continue
            k, v = [x.strip() for x in rule.split(':', 1)]
            camel_k = re.sub(r'-([a-z])', lambda m: m.group(1).upper(), k)
            v_clean = v.replace('"', "'")
            style_obj_parts.append(f'{camel_k}: "{v_clean}"')
        return 'style={{ ' + ', '.join(style_obj_parts) + ' }}'
    jsx = re.sub(r'style="([^"]*)"', style_replacer, jsx)

    # 5. Fix HTML comments
    jsx = re.sub(r'<!--([\s\S]*?)-->', r'{/* \1 */}', jsx)
    return jsx

with open('home.html', 'r', encoding='utf-8') as f:
    home_html = f.read()

with open('faculty.html', 'r', encoding='utf-8') as f:
    faculty_html = f.read()

home_jsx = convert_html_to_jsx(home_html)
faculty_jsx = convert_html_to_jsx(faculty_html)

home_component = f"""import Link from 'next/link';

export default function HomePage() {{
  return (
    <div className="bg-background text-on-surface font-body selection:bg-primary-container selection:text-on-primary-container">
      {home_jsx}
    </div>
  );
}}
"""

faculty_component = f"""import Link from 'next/link';

export default function FacultyPage() {{
  return (
    <div className="bg-background text-on-surface font-body selection:bg-primary-container selection:text-on-primary-container">
      {faculty_jsx}
    </div>
  );
}}
"""

with open('src/app/page.tsx', 'w', encoding='utf-8') as f:
    f.write(home_component)

with open('src/app/faculty/page.tsx', 'w', encoding='utf-8') as f:
    f.write(faculty_component)

print("Pages transformed via python!")
